lattice/schema.py

```diff
@@ -551,6 +624,11 @@ def __init__(self, file_path: pathlib.Path, parent_schema: Schema | None = None)
                 )
         #    else:
         #        raise Exception(f"Unrecognized Object Type, \"{object_type}\" in {self.file_path}")
+        for group in self.data_groups.values():
```

Comment:
Separated the extraction of elements into its own loop, in order to have all DataGroup objects available when any element is populated. This allows for an element's type to find its defining DataGroup and see if custom attributes should be applied based on that DataGroup's parents and/or templates.